packages/mrgn-toasts/src/utils/toast-manager.tsx

```diff
@@ -0,0 +1,143 @@
+import { toast } from "sonner";
+import { WarningToast } from "../components/toasts/warning-toast";
+import { ErrorToast } from "../components/toasts/error-toast";
+import { MultiStepToast } from "../components/toasts/multi-step-toast/multi-step-toast";
+
+export interface MultiStepToastStep {
+  label: string;
+  status: "todo" | "pending" | "success" | "error" | "canceled" | "paused";
+  message?: string;
+  signature?: string;
+  explorerUrl?: string;
+  retry?: () => void;
+}
+
+/** 🔹 Type for MultiStepToastController */
+export interface MultiStepToastController {
+  start: () => void;
+  successAndNext: (explorerUrl?: string, signature?: string) => void;
+  success: (explorerUrl?: string, signature?: string) => void;
+  setFailed: (message?: string, retry?: () => void) => void;
+  pause: () => void;
+  resume: () => void;
+  resetAndStart: () => void;
+  close: () => void;
+}
+
+class ToastManager {
+  showWarningToast(title: string, message: string) {
+    toast(<WarningToast title={title} message={message} />, { duration: Infinity });
+  }
+
+  showErrorToast(title: string, description: string, code?: number) {
+    toast(<ErrorToast title={title} description={description} code={code} />, {
+      duration: Infinity,
+    });
+  }
+
+  createMultiStepToast(title: string, steps: { label: string }[]): MultiStepToastController {
+    const toastId: string = Math.random().toString(36).substring(2, 9);
+
+    const stepsWithStatus: MultiStepToastStep[] = steps.map((step, index) => ({
+      ...step,
+      status: index === 0 ? "pending" : "todo",
+    }));
+
+    toast(<MultiStepToast toastId={toastId} title={title} steps={stepsWithStatus} />, {
+      id: toastId,
+      duration: Infinity,
+    });
+
+    const updateToast = () => {
+      toast(<MultiStepToast toastId={toastId} title={title} steps={stepsWithStatus} />, {
+        id: toastId,
+        duration: Infinity,
+      });
+    };
+
+    const ToastController: MultiStepToastController = {
+      start: () => {
+        updateToast(); 
+      },
+
+      successAndNext: (explorerUrl?: string, signature?: string) => {
```

Comment:
make sure all edgecases are handeled, maybe one could slip through the if statements